File: leetcode/0121-maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0, curMax = 0;
        for (int i = 1; i < prices.length; ++i) {
            curMax += prices[i] - prices[i-1];
            res = Math.max(curMax, res);
            curMax = curMax <= 0 ? 0 : curMax;
        }
        return res;
    }
}

File: leetcode/0121-maxProfit.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        res, cur_max = 0, 0
        for i in range(1, len(prices)):
            cur_max += prices[i] - prices[i-1]
            res = max(cur_max, res)
            cur_max = 0 if cur_max <= 0 else cur_max
        return res
